### PR TITLE
Add UI for viewing review replies

### PR DESCRIPTION
**Pull request by justinshreve**
This PR adds a read-only view of admin replies for reviews. Replies are pulled in and displayed, and a textarea for replying is displayed. The action links and submit button do not do anything yet. This will come in a future PR.

screen shot 2017-09-13 at 4 23 42 pm

cc @kellychoffman @jameskoster since this has UI

See #17693.

To Test:
* Make sure you are using the latest version [of my review branch](https://github.com/woocommerce/wc-api-dev/tree/add/all-product-reviews-endpoint). If you reviewed my previous branches, I have made another adjustment to the API to correctly return formatted review content in the view context. _tl;dr Download the branch again and install this version on your test site._
* Make sure you have a review with a few replies on it. You can reply to a review in wp-admin under each individual product.
* Go to `http://calypso.localhost:3000/store/reviews/:site`.
* Select and expand a review without replies.
* You should see the textarea but nothing else.
* Select and expand a review with replies.
* Replies should load in above the textarea.

### CODE REVIEW

File: client/extensions/woocommerce/app/reviews/review-replies.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	}
+
+	render() {
+		const { replyIds, review, translate } = this.props;
+		const repliesOutput = replyIds.length && replyIds.map( this.renderReply ) || null;
+
+		const textAreaPlaceholder = translate( 'Reply to %(reviewAuthor)s…', {
+			args: { reviewAuthor: review.name }
+		} );
+		const submitButton = 'approved' === review.status ? translate( 'Send reply' ) : translate( 'Approve and send reply' );
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/reviews/review-replies.js]: Hi! I've found a possible matching string that has already been translated **96** times:
`translate( 'Approve and Reply' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by kellychoffman** [comment]
Could we align better with what Comments is doing in the reply text area? Like this:

screen shot 2017-09-13 at 3 24 41 pm

- Placeholder
- Borderless button

**Comment by justinshreve** [comment]
Yup I can do that!

I went back and forth on it since we had [this comment](https://github.com/Automattic/wp-calypso/issues/17047#issuecomment-321565905) and the more recent flow posted for it https://github.com/Automattic/wp-calypso/issues/17693. For Q4 and beyond, we should pick one for-sure canonical source for designs that we know to always be the latest design, otherwise I think it can get confusing during implementation.

**Comment by justinshreve** [comment]
@kellychoffman Done. The textarea currently doesn't grow as you type like the one on comments does. I'll handle that when I finish implementing replying. It does have the correct placeholder text and styling now, with the borderless button.

File: client/extensions/woocommerce/app/reviews/review-replies.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+		/>;
+	}
+
+	render() {
+		const { replyIds, review, translate } = this.props;
+		const repliesOutput = replyIds.length && replyIds.map( this.renderReply ) || null;
+
+		const textAreaPlaceholder = 'approved' === review.status
+			? translate( 'Reply to %(reviewAuthor)s…', { args: { reviewAuthor: review.name } } )
+			: translate( 'Approve and reply to %(reviewAuthor)s…', { args: { reviewAuthor: review.name } } );
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/reviews/review-replies.js]: Hi! I've found a possible matching string that has already been translated **19** times:
`translate( 'Approve and reply to %(commentAuthor)s…' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/reviews/review-reply.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+					<div className="reviews__info">
+						<div className="reviews__author-name">{ reply.author_name }</div>
+						<div className="reviews__date">{ humanDate( reply.date_gmt + 'Z' ) }</div>
+					</div>
+
+					{ this.renderReplyActions() }
+				</div>
+
+				<div className="reviews__reply-content"
+					dangerouslySetInnerHTML={ { __html: content } } //eslint-disable-line react/no-danger
```
Comment (allendav) [inline client/extensions/woocommerce/app/reviews/review-reply.js]: Interesting - this does mean things like img can make it through, e.g.

```
Testing
<b>Bold</b>

<img src="https://upload.wikimedia.org/wikipedia/commons/c/c4/PM5544_with_non-PAL_signals.png" />

<h1>Boo</h1>

<p>Y'all</p>
```

yields

screen shot 2017-09-15 at 1 44 59 pm

Comment (allendav) [reply client/extensions/woocommerce/app/reviews/review-reply.js]: But since the tags allowed are limited, I guess that is OK

**Comment by justinshreve** [comment]
> Please consider limiting the tags permitted in the html before dangerously setting it.

@allendav Would you be okay with us doing this for now to keep reviews on track for Q3? Calypso already uses this in a number of places including normal [comments](https://github.com/Automattic/wp-calypso/blob/6c3abe702b9373d4359530710fa8215c79d6cf8b/client/blocks/comment-detail/comment-detail-comment.jsx#L70) (mentioned below my usage), [plugin descriptions](https://github.com/Automattic/wp-calypso/blob/3af142cea8248bb9c647e988a8e8e8be82c78827/client/my-sites/plugins/plugin-sections/custom.jsx#L35), and more. Like you mentioned, the HTML is limited/filtered on the WP side for what is allowed in comments. 

I'm fine starting the discussion for selecting a library like `sanitize-html-react`, but since we have other usage in Calypso, and it would need to be merged into the root `package.json`, I think that might require some discussion & decision.

**Pull request merged**